engine/src/main/java/org/pentaho/di/trans/steps/edi2xml/grammar/FastSimpleGenericEdifactDirectXMLParser.java

```diff
@@ -0,0 +1,1096 @@
+      throw e;
+    }
+    return retval;
+  }
+
+  // $ANTLR end "data_element"
+
+  public static class data_element_payload_return extends ParserRuleReturnScope {
+    public StringTemplate st;
+
+    public Object getTemplate() {
+      return st;
+    }
+
+    public String toString() {
+      return st == null ? null : st.toString();
+    }
+  };
+
+  // $ANTLR start "data_element_payload"
+  // C:\\workspace-sts\\Kettle trunk -
+  // restruct\\engine\\src\\org\\pentaho\\di\\trans\\steps\\edi2xml\\grammar\\FastSimpleGenericEdifactDirectXML.g:98:1:
+  // data_element_payload : ( composite_data_item ds )* composite_data_item ;
+  public final FastSimpleGenericEdifactDirectXMLParser.data_element_payload_return data_element_payload() throws RecognitionException {
+    FastSimpleGenericEdifactDirectXMLParser.data_element_payload_return retval =
+      new FastSimpleGenericEdifactDirectXMLParser.data_element_payload_return();
+    retval.start = input.LT( 1 );
+
+    try {
+      // C:\\workspace-sts\\Kettle trunk -
+      // restruct\\engine\\src\\org\\pentaho\\di\\trans\\steps\\edi2xml\\grammar\\
+      // FastSimpleGenericEdifactDirectXML.g:98:22:
+      // ( ( composite_data_item ds )* composite_data_item )
+      // C:\\workspace-sts\\Kettle trunk -
+      // restruct\\engine\\src\\org\\pentaho\\di\\trans\\steps\\edi2xml\\grammar\\
+      // FastSimpleGenericEdifactDirectXML.g:98:24:
+      // ( composite_data_item ds )* composite_data_item
+
+      buf.append( TAG_ELEMENT );
+      // C:\\workspace-sts\\Kettle trunk -
+      // restruct\\engine\\src\\org\\pentaho\\di\\trans\\steps\\edi2xml\\grammar\\
+      // FastSimpleGenericEdifactDirectXML.g:99:4:
+      // ( composite_data_item ds )*
+      loop5: do {
+        int alt5 = 2;
+        int LA5_0 = input.LA( 1 );
+
+        if ( ( LA5_0 == TEXT_DATA ) ) {
+          int LA5_1 = input.LA( 2 );
+
+          if ( ( LA5_1 == COMPLEX_ELEMENT_ITEM_SEPARATOR ) ) {
+            alt5 = 1;
+          }
+
+        } else if ( ( LA5_0 == COMPLEX_ELEMENT_ITEM_SEPARATOR ) ) {
+          alt5 = 1;
+        }
+
+        switch ( alt5 ) {
+          case 1:
+            // C:\\workspace-sts\\Kettle trunk -
+            // restruct\\engine\\src\\org\\pentaho\\di\\trans\\steps\\edi2xml\\grammar\\
+            // FastSimpleGenericEdifactDirectXML.g:99:5:
+            // composite_data_item ds
+
+            pushFollow( FOLLOW_composite_data_item_in_data_element_payload160 );
+            composite_data_item();
+            state._fsp--;
+            pushFollow( FOLLOW_ds_in_data_element_payload162 );
+            ds();
+            state._fsp--;
+            break;
+
+          default:
+            break loop5;
+        }
+      } while ( true );
+      pushFollow( FOLLOW_composite_data_item_in_data_element_payload166 );
+      composite_data_item();
+      state._fsp--;
+      buf.append( TAG_ELEMENT_END );
+      retval.stop = input.LT( -1 );
+
+    } catch ( RecognitionException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)